src/Ingest.php

```diff
@@ -133,8 +135,8 @@ private function waitForAcknowledgment($stream): void
             }
 
             $response .= $part;
-
-        } while (strlen($response) < 4);
+            $attempts++;
+        } while (strlen($response) < 4 && ! feof($stream) && $attempts < 5);
```

Comment:
Rather than returning `false` when the stream has reached `eof`, `fread` will return an empty string.

This change ensure we only attempt to read from an open stream.

I found this bug when testing against the agent restart functionality. The loop would continue indefinitely.

Although I don't have an exact scenario where it is required, I also added a manual limit on the number of times we attempt to read from the stream. This felt safest in case there are other situations where this loop could enter an infinite loop.